Feature/electricity service

Daan298261 opened the pull request:
full commit

Daan298261 committed:
checkin gitignore

Daan298261 committed:
init slution c#

Daan298261 committed:
Full commit

markieo1 commented on .gitignore:
```diff
@@ -0,0 +1,261 @@
+## Ignore Visual Studio temporary files, build results, and
```
Please update this file to the latest version on [VisualStudio.gitignore](https://github.com/github/gitignore/blob/master/VisualStudio.gitignore)

markieo1 commented on ElectricityService.App/Properties/launchSettings.json:
```diff
@@ -0,0 +1,10 @@
+{
```
File should not exist in the repository, can be fixed by updating the gitignore

Daan298261 replied:
did so, @Aplicticus can verify Still VS insists that the file should be committed...

markieo1 commented on ElectricityService.Core/Models/Ship.cs:
```diff
@@ -0,0 +1,19 @@
+        public string Name { get; set; }
+
+        [IgnoreDataMember]
+        public ShipService ShipService { get; set; }
```
This should not be on a ship?

Daan298261 replied:
you said 'look in my project for it' so i did.

markieo1 commented on ElectricityService.Infrastructure/Services/ElectricityService.cs:
```diff
@@ -0,0 +1,36 @@
+                ShipService shipService = new ShipService();
+                shipService.Name = "Electricity";
+                shipService.Id = new Guid();
+                shipService.Price = 2500;
```
The electricity service does not decide the price!

Daan298261 replied:
according to the data that kevin requires it does

markieo1 commented on README.md:
```diff
@@ -1 +0,0 @@
-# ElectricityService
```
No need for this change.

Daan298261 committed:
je gitgonore.
last commit

Daan298261 committed:
Updated ElectricityService with repository and numerous vanityFixes.

Note: Even tough launchsettings.json should not be committed (as per the gitignore file) Visual studio insists on including the file.

Remove it later if you wish

Daan298261 committed:
added guid in readme for easy access to test the events

Daan298261 committed:
removed launcsettings

Daan298261 committed:
updated travisfile

Daan298261 committed:
because people rather want to search the entire project for a damn guid.

markieo1 commented on ElectricityService.Infrastructure/DI/DIHelper.cs:
```diff
@@ -0,0 +1,37 @@
+            services.AddTransient<IShipRepository, ShipRepository>();
+            services.AddTransient<IElectricityService, Services.ElectricityService>();
+
+            services.AddSingleton<IEventHandler, RMQMessageHandler>((provider) => new RMQMessageHandler(configuration.GetSection("AMQP").Value));
```
This causes issues on production see [App doesnt start in production](https://github.com/GiethoornInternationalHarbor/RefuelService/issues/2)

Daan298261 replied:
has been fixed in new branch called hotfix/AMQP_URL. i am closing this PR and opening a new one on that branch. it should launch without any issues now.

Daan298261 committed:
Fixed AMQP environment variable to fix DO crash

Daan298261 commented:
Migrated to branch hotfix/AMQP_URL

markieo1 commented:
Please merge that branch in to this one. Then we do not lose alll the comments ;-)

Daan298261 committed:
Merge pull request #3 from GiethoornInternationalHarbor/hotfix/AMQP_URL

Fixed AMQP environment variable to fix DO crash

Daan298261 commented:
Merged hotfix into this as per your request
